src/main/java/io/cdap/plugin/successfactors/common/util/ExceptionParser.java

```diff
@@ -0,0 +1,176 @@
+   *
+   * @param msg               stage wise error message for example the caller is
+   *                          - failure while testing OData URL
+   *                          - failure while reading metadata
+   *                          - failure while reading total available record count
+   *                          - failure while reading actual data
+   * @param responseContainer {@code SuccessFactorsResponseContainer} contains the http response details after calling
+   *                          SAP SuccessFactors service
+   * @throws SuccessFactorsServiceException in case of any error scenario, it prepares and throw this exception.
+   */
+  public static void checkAndThrowException(String msg, SuccessFactorsResponseContainer responseContainer)
+    throws SuccessFactorsServiceException {
+
+    String failureMessage = msg;
+    SuccessFactorsError error = null;
+
+    if (responseContainer.getHttpStatusCode() == HttpURLConnection.HTTP_UNAUTHORIZED) {
+      failureMessage += ResourceConstants.ROOT_CAUSE_LOG.getMsgForKey()
+        .concat(ResourceConstants.ERR_INVALID_CREDENTIAL.getMsgForKey());
+
+      throw new SuccessFactorsServiceException(failureMessage, responseContainer.getHttpStatusCode());
+    } else if (responseContainer.getHttpStatusCode() == HttpURLConnection.HTTP_NOT_FOUND) {
+      failureMessage += ResourceConstants.ROOT_CAUSE_LOG.getMsgForKey()
+        .concat(ResourceConstants.ERR_INVALID_ENTITY_NAME.getMsgForKey());
+
+      throw new SuccessFactorsServiceException(failureMessage, responseContainer.getHttpStatusCode());
+    } else if (responseContainer.getHttpStatusCode() != HttpURLConnection.HTTP_OK) {
+      String rawResponseString = "";
+      InputStream rawStream = responseContainer.getResponseStream();
+      if (rawStream != null) {
+        try (BufferedReader bufferedReader = new BufferedReader(
+          new InputStreamReader(rawStream, StandardCharsets.UTF_8))) {
+
+          rawResponseString = bufferedReader.lines().collect(Collectors.joining(" "));
+          // For handling broken hyperlinks to SAP docs in error messages returned by SAP SuccessFactors OData v2 API
+          if (rawResponseString.contains("Refer to https")) {
+            rawResponseString = rawResponseString.substring(0, rawResponseString.indexOf("Refer to https"));
+          }
+        } catch (IOException ioe) {
+          // no-ops
+          LOG.error("Exception thrown while reading input stream ", ioe);
+        }
+      }
+
+      LOG.error("HTTP Code: {}", responseContainer.getHttpStatusCode());
+      LOG.error("Detailed error message: {} {}", msg, rawResponseString);
+
+      try {
+        error = GSON.fromJson(rawResponseString, SuccessFactorsError.class);
+      } catch (JsonSyntaxException | JsonIOException je) {
+        // html errors are only found in case of invalid SuccessFactors service namespace
+        if (rawResponseString.startsWith("<html>")) {
```

Comment:
create a variable

Reply:
Added `HTML_TAG` for` "<html>"`  4dbd00c9806f47df7f78cf411834bc8c7a613f94